save-api/src/main/kotlin/org/cqfn/save/api/AutomaticTestInitializator.kt

```diff
@@ -0,0 +1,254 @@
+            processAdditionalFiles(webClientProperties.fileStorage, it)
+        }
+
+        if (evaluatedToolProperties.additionalFiles != null && additionalFileInfoList == null) {
+            return
+        }
+
+        val msg = additionalFileInfoList?.let {
+            "with additional files: ${additionalFileInfoList.map { it.name }}"
+        } ?: {
+            "without additional files"
+        }
+        log.info("Starting submit execution $msg, type: $executionType")
+
+        val (organization, executionRequest) = submitExecution(executionType, additionalFileInfoList) ?: return
+
+        // Sending requests, which checks current state, until results will be received
+        // TODO: in which form do we actually need results?
+        val resultExecutionDto = getExecutionResults(executionRequest, organization.id!!)
+        val resultMsg = resultExecutionDto?.let {
+            "Execution is finished with status: ${resultExecutionDto.status}. " +
+                    "Passed tests: ${resultExecutionDto.passedTests}, failed tests: ${resultExecutionDto.failedTests}, skipped: ${resultExecutionDto.skippedTests}"
+        } ?: "Some errors occurred during execution"
+
+        log.info(resultMsg)
+    }
+
+    /**
+     * Submit execution according [executionType]
+     *
+     * @param executionType
+     * @param additionalFiles
+     * @return pair of organization and submitted execution request
+     */
+    private suspend fun submitExecution(
+        executionType: ExecutionType,
+        additionalFiles: List<FileInfo>?
+    ): Pair<Organization, ExecutionRequestBase>? {
+        val (organization, executionRequest) = if (executionType == ExecutionType.GIT) {
+            buildExecutionRequest()
+        } else {
+            val userProvidedTestSuites = verifyTestSuites() ?: return null
+            buildExecutionRequestForStandardSuites(userProvidedTestSuites)
+        }
+        httpClient.submitExecution(executionType, executionRequest, additionalFiles)
+        return organization to executionRequest
+    }
+
+    /**
+     * Build execution request for git mode according provided configuration
+     *
+     */
+    private suspend fun buildExecutionRequest(): Pair<Organization, ExecutionRequest> {
+        val (organization, project) = getOrganizationAndProject()
+
+        val gitDto = GitDto(
+            url = evaluatedToolProperties.gitUrl,
+            username = evaluatedToolProperties.gitUserName,
+            password = evaluatedToolProperties.gitPassword,
+            branch = evaluatedToolProperties.branch,
+            hash = evaluatedToolProperties.commitHash
+        )
+
+        // executionId will be calculated at the server side
+        val executionId = null
+
+        return organization to ExecutionRequest(
+            project = project,
+            gitDto = gitDto,
+            testRootPath = evaluatedToolProperties.testRootPath,
+            sdk = Jdk("11"),
```

Comment:
Should SDK be configurable too?

Reply:
Yes, I made the corresponding `TODO` in code, however I didn't implemented some clever mapping for now

I will do this in future PR's